### PR TITLE
backport v1.12: IPsec upgrade tests

### PR DESCRIPTION
**Pull request by brb**
Manual backport of https://github.com/cilium/cilium/pull/26983 and https://github.com/cilium/cilium/pull/27150.

### CODE REVIEW

**Comment by brlbil** [comment]
It seems added workflow is not run.

**Review by brlbil — APPROVED** [review]

**Review by jschwinger233 — APPROVED** [review]

File: .github/workflows/tests-ipsec-upgrade.yaml [hunk 1]
```diff
@@ -107,7 +106,7 @@ jobs:
             tunnel: 'vxlan'
             encryption: 'ipsec'
             encryption-node: 'false'
-            endpoint-routes: 'true'
+            endpoint-routes: 'false' # Due to https://github.com/cilium/cilium/pull/22333
```
Comment (pchaigno) [inline .github/workflows/tests-ipsec-upgrade.yaml:109]: Should this be in its own commit? Or in the first commit with a backporter note?

Comment (brb) [reply .github/workflows/tests-ipsec-upgrade.yaml:109]: :+1: Added the note in the first commit. Same for the https://github.com/cilium/cilium/pull/27174 PR.

**Review by pchaigno — APPROVED** [review]
